pkg/utils/common/controller.go

```diff
@@ -229,6 +229,20 @@ func (s *SuiteController) DeleteConfigMap(name, namespace string) error {
 	return s.KubeInterface().CoreV1().ConfigMaps(namespace).Delete(context.TODO(), name, metav1.DeleteOptions{})
 }
 
+//CheckIfNamespaceExists - find if a namespace exists
+func (h *SuiteController) CheckIfNamespaceExists(name string) bool {
+
+	// Check if the E2E test namespace already exists
+	_, err := h.KubeInterface().CoreV1().Namespaces().Get(context.TODO(), name, metav1.GetOptions{})
+
+	if err != nil && !k8sErrors.IsNotFound(err) {
```

Comment:
Leaving this to Flavius as I don't know how this project is code, but this function looks wrong. First becase all the lines from this one to the bottom can be replaced with:
```
return err == nil
```
But the main problem here is that you are swallowing an error silently. Maybe you couldn't fetch the namespace for some random reason and it exists (`err != ni && !k8sErrors.IsNotFound(err)` -> `return false`).